From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Wed, 12 Oct 2022 10:42:15 -0400
Subject: [PATCH] Leaf config

Leaf Config v3
including load config, backup old or outdated config, and add config to spark profiler automatically.

diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 455302161c8b24a1c953005e0597f3c69f764e1e..69719421061869f145b66ac470957b9fa24445e6 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -102,6 +102,8 @@ public class Main {
                 writePidFile(pidFilePath);
             }
 
+            org.dreeam.leaf.config.LeafConfig.loadConfig(); // Leaf - Leaf config
+
             // Paper start - Perf: Run CrashReport preload asynchronously
             // The actual result is discarded/irrelevant, but hardware and software info is memoized
             Thread.ofPlatform().daemon().name("CrashReport preload thread").start(CrashReport::preload);
@@ -127,6 +129,7 @@ public class Main {
             Thread.ofPlatform().daemon().name("DataFixers init thread").start(DataFixers::getDataFixer);
             // Paper end - Perf: Init DataConverter asynchronously
             Util.startTimerHackThread();
+            org.dreeam.leaf.config.ConfigModule.loadAfterBootstrap(); // Leaf - Leaf config - post load
             Path settingsFile = Paths.get("server.properties");
             DedicatedServerSettings settings = new DedicatedServerSettings(options); // CraftBukkit - CLI argument support
             settings.forceSave();
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 165b1e26c23eeedf1aa71eb2cc8827fd744ee820..bf5bf3d2a03401a8d7ac91b91b43cc9df0185f47 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1287,6 +1287,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             org.spigotmc.WatchdogThread.tick();
             // Paper end
             org.spigotmc.WatchdogThread.hasStarted = true; // Paper
+            org.dreeam.leaf.config.LeafConfig.regSparkExtraConfig(); // Leaf - Leaf config
             // Paper start - Add onboarding message for initial server start
             if (io.papermc.paper.configuration.GlobalConfiguration.isFirstStart) {
                 LOGGER.info("*************************************************************************************");
